List/ArrayList.cs

```diff
@@ -4,6 +4,497 @@ namespace List
+                }
+                else
+                {
+                    ++index;
+                }
+            }
+        }
+
+        public void RemoveAllByValue(int value)
+        {
+            int index = 0;
+
+            while (index < Length)
+            {
+                if (_array[index] == value)
+                {
+                    RemoveAtIndex(index);
+                }
+                else
+                {
+                    ++index;
+                }
+            }
+        }
+
+        public void AddArrayList(ArrayList arrayList)
+        {
+            AddArrayListAtIndex(arrayList, Length);
+        }
+
+        public void AddArrayListToStart(ArrayList arrayList)
+        {
+            AddArrayListAtIndex(arrayList, 0);
+        }
+
+        public void AddArrayListAtIndex(ArrayList arrayList, int index)
+        {
+            if (index < 0 || index > Length)
+            {
+                throw new IndexOutOfRangeException();
+            }
+            else if (arrayList.Length == 0)
+            {
+                return;
+            }
+
+            Length = Length + arrayList.Length;
+
+            CheckAndChangeSize();
+
+            int count = 0;
+
+            for (int i = index; i < Length; i++)
+            {
+                if (i + arrayList.Length < _array.Length)
+                {
+                    _array[i + arrayList.Length] = _array[i];
+                }
+                if (count < arrayList.Length)
+                {
+                    _array[i] = arrayList[count];
+                }
+
+                ++count;
+            }
+        }
+
+        public override string ToString()
+        {
+            string stringResult = "";
+
+            for (int i = 0; i < Length; i++)
+            {
+                if (i == Length - 1)
+                {
+                    stringResult += _array[i];
+                }
+                else
+                {
+                    stringResult += _array[i] + " ";
+                }
+            }
+
+            return stringResult;
+        }
+
+        public override bool Equals(object obj)
+        {
+            ArrayList list = (ArrayList)obj;
+
+            bool result = true;
+
+            if (this.Length != list.Length)
+            {
+                result = false;
+            }
+            for (int i = 0; i < Length; i++)
+            {
+                if (this._array[i] != list._array[i])
+                {
+                    result = false;
+                }
+            }
+
+            return result;
+        }
+
+        private void CheckAndChangeSize()
+        {
+            if (Length >= _array.Length)
+            {
+                while (Length >= _array.Length)
+                {
+                    UpSize();
+                }
+            }
+            else if (Length <= _array.Length / 2 + 1)
+            {
+                DownSize();
+            }
+        }
+
+        private void UpSize()
```

Comment:
Try to use `UpdateSize` which will make something like:
`var newSize = Length==Count ? 1.33 : 0.67;`
Instead of two methods